File: 1-Iniciante/1021-Notas-e-Moedas/1021.py
din = {100 : 0, 50: 0, 20: 0, 10: 0, 5: 0, 2: 0, 1:0, 0.5: 0, 0.25: 0, 0.1: 0, 0.05: 0, 0.01: 0}
n = float(input())
if n >= 100:
    din[100] = int(n // 100)
    n -= 100 * (n // 100)
    n = round (n, 2)
if n >= 50:
    din[50] = int(n // 50)
    n -= 50 * (n // 50)
    n = round (n, 2)
if n >= 20:
    din[20] = int(n // 20)
    n -= 20 * (n // 20)
    n = round (n, 2)
if n >= 10:
    din[10] = int(n // 10)
    n -= 10 * (n // 10)
    n = round (n, 2)
if n >= 5:
    din[5] = int(n // 5)
    n -= 5 * (n // 5)
    n = round (n, 2)
if n >= 2:
    din[2] = int(n // 2)
    n -= 2 * (n // 2)
    n = round (n, 2)
if n >= 1:
    din[1] = int(n)
    n -= int(n)
    n = round (n, 2)
if n >= 0.5:
    din[0.5] = int(n / 0.5)
    n -= 0.5 * (n // 0.5)
    n = round (n, 2)
if n >= 0.25:
    din[0.25] = int(n / 0.25)
    n -= 0.25 * (n // 0.25)
    n = round (n, 2)
if n >= 0.1:
    din[0.1] = int(n / 0.1)
    n -= 0.1 * (n // 0.1)
    n = round (n, 2)
if n >= 0.05:
    din[0.05] = int(n / 0.05)
    n -= 0.05 * (n // 0.05)
    n = round (n, 2)
if n >= 0.01:
    din[0.01] = int(n / 0.01)
print ("NOTAS:")
print (din[100], "nota(s) de R$ 100.00")
print (din[50], "nota(s) de R$ 50.00")
print (din[20], "nota(s) de R$ 20.00")
print (din[10], "nota(s) de R$ 10.00")
print (din[5], "nota(s) de R$ 5.00")
print (din[2], "nota(s) de R$ 2.00")
print ("MOEDAS:")
print (din[1], "moeda(s) de R$ 1.00")
print (din[0.5], "moeda(s) de R$ 0.50")
print (din[0.25], "moeda(s) de R$ 0.25")
print (din[0.1], "moeda(s) de R$ 0.10")
print (din[0.05], "moeda(s) de R$ 0.05")
print (din[0.01], "moeda(s) de R$ 0.01")

File: 1-iniciante/_1021.java
import java.util.Scanner;
//
public class Main{
	public static void main(String args[]){
		Scanner scan=new Scanner(System.in);
		double x=scan.nextDouble();
		System.out.printf("NOTAS:%n%.0f nota(s) de R$ 100.00%n%.0f nota(s) de R$ 50.00%n%.0f nota(s) de R$ 20.00%n%.0f nota(s) de R$ 10.00%n%.0f nota(s) de R$ 5.00%n%.0f nota(s) de R$ 2.00%n",x/100-x%100/100,x%100/50-x%100%50/50,x%100%50/20-x%100%50%20/20,x%100%50%20/10-x%100%50%20%10/10,x%100%50%20%10/5-x%100%50%20%10%5/5,x%100%50%20%10%5/2-x%100%50%20%10%5%2/2);
		x=x%100%50%20%10%5%2;
		System.out.printf("MOEDAS:%n%.0f moeda(s) de R$ 1.00%n%.0f moeda(s) de R$ 0.50%n%.0f moeda(s) de R$ 0.25%n%.0f moeda(s) de R$ 0.10%n%.0f moeda(s) de R$ 0.05%n%.0f moeda(s) de R$ 0.01%n",x-x%1,x%1/0.5-x%1%0.5/0.5,x%1%0.5/0.25-x%1%0.5%0.25/0.25,x%1%0.5%0.25/0.1-x%1%0.5%0.25%0.1/0.1,x%1%0.5%0.25%0.1/0.05-x%1%0.5%0.25%0.1%0.05/0.05,x%1%0.5%0.25%0.1%0.05/0.01);
	}
}
